client/state/plugins/wporg/test/test-selectors.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { assert } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import selectors from '../selectors';
+
+const state = {
+	test: Object.freeze( { slug: 'test' } ),
+	fetchingTest: Object.freeze( { slug: 'fetchingTest', isFetching: true } )
+};
+
+describe( 'WPorg Selectors', function() {
+	before( function() {
+	} );
+
+	after( function() {
+	} );
+
+	it( 'Should contain getPlugin method', function() {
+		assert.equal( typeof selectors.getPlugin, 'function' );
+	} );
+
+	it( 'Should contain isFetching method', function() {
+		assert.equal( typeof selectors.isFetching, 'function' );
+	} );
+
+	describe( 'getPlugin', function() {
+		it( 'Should get null if the requested plugin is not in the current state', function() {
+			assert.equal( selectors.getPlugin( state, 'no-test' ), null );
+		} );
+
+		it( 'Should get the plugin if the requested plugin is in the current state', function() {
+			assert.equal( selectors.getPlugin( state, 'test' ).slug, 'test' );
+		} );
+
+		it( 'Should return a new object with no pointers to the one stored in state', function() {
+			let plugin = selectors.getPlugin( state, 'test' );
+			plugin.isFetching = true;
+			assert.equal( selectors.getPlugin( state, 'test' ).isFetching, undefined );
+		} );
```

Comment:
Previously the .org Plugin store returned 3 possible values for getPlugin( undefiend, null, and an object )
Do we need to add more tests to here? I think it would make sense to trigger a request to fetch the data from the org store if we getPlugin function returned undefined. 

I see that now we moved the fetching of the data to the 2 different components. ( main.jsx and plugin.jsx ) 
Is that the preferred way of doing things in Redux? 

Looking at the redux example in redux maybe we need some sort of `fetchPluginIfNeeded` method. 
That figures out if we need to fetch a plugin from .org or not See http://redux.js.org/docs/advanced/ExampleRedditAPI.html#action-creators-and-constants